torchtitan/utils.py

```diff
@@ -106,7 +106,8 @@ def set_determinism(
 
     # The native RNGs and python RNG may not be important, except for the 1-D PP case, but we seed them for consistency.
     torch.manual_seed(seed)
-    os.environ["PYTHONHASHSEED"] = str(seed)
+    # PYTHONHASHSEED can be a decimal number in the range [0,2**32 - 1]
+    os.environ["PYTHONHASHSEED"] = str(seed % (2**32 - 1))
```

Comment:
I guess if you `% N`, then the value is in `[0, N-1]`.

Reply:
oops, thanks for catching!